src/backend/app.py

```diff
@@ -11,25 +16,123 @@
 def login():
     """Handles login routing"""
     user_login = Login()
-    user = request.json["user"]
-    password = request.json["password"]
-    if user_login.login(user, password):
-        return f"welcome {user}", 200
-    return "User not found, please try again", 404
+    json_form = request.get_json(force=True)
+
+    if isinstance(json_form, dict):
+        user = json_form.get("user", "")
+        password = json_form.get("password", "")
+        if user_login.login(user, password):
+            return f"welcome {user}", 200
+        return "User not found, please try again", 401
+    return "", 400
 
 
 @app.route("/register", methods=["POST", "GET"])
 def register():
     """Handles register routing"""
     user_login = Login()
-    username = request.json["username"]
-    email = request.json["email"]
-    password = request.json["password"]
-    phone = request.json["phone"]
-    result = user_login.register(username, email, password, phone)
-    if not result.status:
-        return result.message, 400
-    return result.message, 200
+    json_form = request.get_json(force=True)
+
+    if isinstance(json_form, dict):
+        username = json_form.get("username", "")
+        email = json_form.get("email", "")
+        password = json_form.get("password", "")
+        phone = json_form.get("phone", "")
+        result = user_login.register(username, email, password, phone)
+        if not result.status:
+            return result.message, 400
+        return result.message, 201
+    return "", 400
+
+
+@app.route("/", methods=["POST", "GET"])
+@app.route("/main", methods=["POST", "GET"])
+def mainpage():
+    """Handle mainpage requests"""
+    mainpage_obj = MainPage()
+    if request.method == "POST":
+        return mainpage_post(mainpage_obj)
+
+    args = request.args
+    return mainpage_get(mainpage_obj, args)
+
+
+def mainpage_get(mainpage_obj: MainPage, args: MultiDict):
+    """Helper for mainpage get requests"""
+    action_type = namedtuple(
```

Comment:
obviously it's a helper, but what does it do?

Reply:
It processes every possible get requests to main, as per the docstring

Reply:
I suggest you list the cases

Reply:
Aight